compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/optimizer/LargeMethodOptimizer.java

```diff
@@ -104,28 +122,1063 @@ public void splitLargeBIRFunctions(BIRPackage birPkg) {
+                setArrayElement(newInsList, handleArrayOperand, currOperand, listConstructorEntry, arrayIndex,
+                        tempVars);
+            }
+            // no need to set the below in else part as it will not affect since splitNow is considered along with this
+            splitFuncEnv.splitHere = splitPointDetails.splitHere;
+            return true;
+        }
+        return false;
+    }
+
+    private void setMapElement(List<BIRNonTerminator> newInsList,
+                                 BIROperand handleArrayOperand, BIROperand valueOrExprOperand,
+                                 BIRNode.BIRMappingConstructorEntry mappingConstructorEntry, int arrayIndex,
+                                 TempVarsForArraySplit tempVars) {
+        JMethodCallInstruction callSetEntry = new JMethodCallInstruction(null);
+        callSetEntry.invocationType = INVOKESTATIC;
+        callSetEntry.jClassName = LARGE_STRUCTURE_UTILS;
+        tempVars.tempVarsUsed = true;
+        BIRNonTerminator.ConstantLoad loadArrayIndex = new BIRNonTerminator.ConstantLoad(null,
+                (long) arrayIndex, symbolTable.intType, tempVars.arrayIndexOperand);
+        newInsList.add(loadArrayIndex);
+
+        BIRNonTerminator.TypeCast typeCastInstruction = new BIRNonTerminator.TypeCast(null,
+                tempVars.typeCastOperand, valueOrExprOperand, symbolTable.anyOrErrorType, true);
+        newInsList.add(typeCastInstruction);
+
+        if (mappingConstructorEntry instanceof BIRNode.BIRMappingConstructorKeyValueEntry) {
+            callSetEntry.jMethodVMSig = "(" + GET_HANDLE_VALUE + GET_OBJECT + GET_OBJECT + "J)V";
+            callSetEntry.args = new ArrayList<>(Arrays.asList(handleArrayOperand,
+                    ((BIRNode.BIRMappingConstructorKeyValueEntry) mappingConstructorEntry).keyOp,
+                    tempVars.typeCastOperand, tempVars.arrayIndexOperand));
+            callSetEntry.name = "setKeyValueEntry";
+        } else {
+            callSetEntry.jMethodVMSig = "(" + GET_HANDLE_VALUE + GET_OBJECT + "J)V";
+            callSetEntry.args = new ArrayList<>(Arrays.asList(handleArrayOperand, tempVars.typeCastOperand,
+                    tempVars.arrayIndexOperand));
+            callSetEntry.name = "setSpreadFieldEntry";
+        }
+        newInsList.add(callSetEntry);
+    }
+
+    private void setArrayElement(List<BIRNonTerminator> newInsList,
+                                 BIROperand handleArrayOperand, BIROperand insLhsOp,
+                                 BIRNode.BIRListConstructorEntry listConstructorEntry, int arrayIndex,
+                                 TempVarsForArraySplit tempVars) {
+        JMethodCallInstruction callSetEntry = new JMethodCallInstruction(null);
+        callSetEntry.invocationType = INVOKESTATIC;
+        callSetEntry.jClassName = LARGE_STRUCTURE_UTILS;
+        callSetEntry.jMethodVMSig = "(" + GET_HANDLE_VALUE + GET_OBJECT + "J)V";
```

Comment:
Shall we move `"(" + GET_HANDLE_VALUE + GET_OBJECT + "J)V"` signature to a constant and reuse it?